protocol/contracts/amo/Ramos.sol

```diff
@@ -191,207 +264,304 @@ contract RAMOS is Ownable, Pausable {
+        // Borrow the protocol token
+        protocolTokenVault.borrowProtocolToken(protocolTokenAmountIn, address(this));
+
+        // Collect the fees from the input protocol token amount
+        uint256 feeAmt = protocolTokenAmountIn * rebalanceFees.rebalanceJoinFeeBps / BPS_PRECISION;
+        if (feeAmt != 0) {
+            protocolToken.safeTransfer(feeCollector, feeAmt);
+        }
 
-        uint256 stableTokenIndex = templeBalancerPoolIndex == 0 ? 1 : 0;
-        uint256 amountOut = poolHelper.exitPool(
-            bptAmountIn, minAmountOut, rebalancePercentageBoundLow, rebalancePercentageBoundUp,
-            postRebalanceSlippage, stableTokenIndex, templePriceFloorNumerator, stable
+        // Send the balance to the poolHelper
+        uint256 joinAmountIn = protocolTokenAmountIn - feeAmt;
+        protocolToken.safeTransfer(address(poolHelper), joinAmountIn);
+
+        // protocolToken single side join
+        uint256 bptOut = poolHelper.joinPool(
+            joinAmountIn, minBptOut, rebalancePercentageBoundUp,
+            rebalancePercentageBoundLow, treasuryPriceIndex(), 
+            postRebalanceSlippage, protocolTokenBalancerPoolIndex, protocolToken
         );
+        emit RebalanceDownJoin(protocolTokenAmountIn, bptOut);
 
-        lastRebalanceTimeSecs = uint64(block.timestamp);
-        emit WithdrawStable(bptAmountIn, amountOut);
+        // deposit and stake BPT
+        bptToken.safeTransfer(address(amoStaking), bptOut);
+        amoStaking.depositAndStake(bptOut);
     }
 
     /**
-     * @notice Add liquidity with both TEMPLE and stable tokens into balancer pool. 
-     * Treasury Price Floor is expected to be within bounds of multisig set range.
+     * @notice Get the quote used to add liquidity proportionally
+     * @dev Since this is not the view function, this should be called with `callStatic`
+     */
+    function proportionalAddLiquidityQuote(
+        uint256 quoteTokenAmount,
+        uint256 slippageBps
+    ) external override returns (
+        uint256 protocolTokenAmount,
+        uint256 expectedBptAmount,
+        uint256 minBptAmount,
+        IBalancerVault.JoinPoolRequest memory requestData
+    ) {
+        return poolHelper.proportionalAddLiquidityQuote(quoteTokenAmount, slippageBps);
+    }
+
+    /**
+     * @notice Add liquidity with both `protocolToken` and `quoteToken` into balancer pool. 
+     * TPI is expected to be within bounds of multisig set range.
      * BPT tokens are then deposited and staked in Aura.
      * @param request Request data for joining balancer pool. Assumes userdata of request is
      * encoded with EXACT_TOKENS_IN_FOR_BPT_OUT type
-     * @param minBptOut Minimum amount of BPT tokens expected to receive
      */
     function addLiquidity(
-        AMO__IBalancerVault.JoinPoolRequest memory request,
-        uint256 minBptOut
-    ) external onlyOwner {
+        IBalancerVault.JoinPoolRequest memory request
```

Comment:
nit: gas savings: we could use `calldata` here as request is not modified

Reply:
Hmm - there's a really strange bytesize (ie `yarn size`) issue if I change this to calldata. 

as `memory`:   ` |  Ramos                            ·      23.776  ·                |`
as `calldata`: ` |  Ramos                            ·      24.541  ·        +0.765  │`

So we're right up close to the size limit as it is, and then calldata blows past it. So while it's a small gas saving, I don't think we can use it.

Reply:
ah I see, now that reminds me why i left it initially as memory, my bad. thanks!

Reply:
It's a bizarre side effect. Gheeze the solidity compiler could do with some work!